web/src/main/java/com/marklogic/hub/web/service/ArtifactService.java

```diff
@@ -0,0 +1,36 @@
+package com.marklogic.hub.web.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.hub.HubConfig;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ArtifactService {
```

Comment:
Is there any reason to have this class other than similar classes were created in the past? To me, it seems sufficient for the web controller - or whatever class receives the HTTP request - to talk directly to  the DataService. And that controller can be responsible for instantiating the DS - i.e. it would depend on HubConfig, and then it would use that to create a new DS instance on the fly.

Reply:
I think not having this web/service class also moves us away from everything being a singleton based on a single HubConfig instance. In the above pattern, the controller would obtain a HubConfig based on the current user (i.e. from the user's HTTP session) and then instantiate the appropriate DS on the fly.

Reply:
Thinned it down by calling the Data Services class directly. I did add an abstract artifact controller to simplify implementations for future artifacts.